nextcord/application_command.py

```diff
@@ -745,12 +745,50 @@ def from_callback(
                 skip_counter = 1
                 # TODO: use typing.get_type_hints when 3.9 is standard
                 typehints = typing_extensions.get_type_hints(self.callback, include_extras=True)
+                callback_params = signature(self.callback).parameters
+                possible_options: Set[str] = set()
+
+                for i, param in enumerate(callback_params.values()):
+                    if not (
+                        param.annotation is param.empty  # could be a self or interaction parameter
+                        or issubclass(
+                            param.annotation, Interaction
+                        )  # will always be an interaction parameter
+                        # TODO: use typing.Self when 3.11 is standard
+                        or param.annotation
+                        is typing_extensions.Self  # will always be a self parameter
+                        or isinstance(
+                            param.annotation, typing.TypeVar
+                        )  # will always be a self parameter
+                    ) and i in (0, 1):
+                        possible_options.add(param.name)
+
+                intersection = {param.name for param in callback_params.values()} - possible_options
+
+                if self.parent_cog is not None and len(intersection) < 2:
+                    raise ValueError(
+                        f"Callback {self.error_name} is missing the self and/or interaction parameters. Please double check your function definition."
+                    )
+                elif len(intersection) < 1:
+                    raise ValueError(
+                        f"Callback {self.error_name} is missing the interaction parameter. Please double check your function definition."
+                    )
+
+                # not needed later on
+                del possible_options, intersection
+
                 # Getting the callback with `self_skip = inspect.ismethod(self.callback)` was problematic due to the
                 #  decorator going into effect before the class is instantiated, thus being a function at the time.
                 #  Try to look into fixing that in the future?
                 #  If self.parent_cog isn't reliable enough, we can possibly check if the first parameter name is `self`
-                if self.parent_cog:
-                    skip_counter += 1
+                self_skip = bool(self.parent_cog)
+                first_arg = True
+                for name, param in callback_params.items():
+                    if first_arg:
+                        if not self_skip:
+                            first_arg = False
+                        else:
+                            self_skip = False
```

Comment:
I'm not sure I see what this is doing. It appears `first_arg` and `self_skip` are not referenced anywhere else in the rest of this function, so it seems this code is unused.

Reply:
I thought this was old code? Not sure
I'll remove it anyways